src/commands/accounts/remove.ts

```diff
@@ -37,30 +37,25 @@ export default class AccountsRemove extends BaseCommand<typeof AccountsRemove> {
    *
    * @returns Empty promise
    */
-  public async run(): Promise<void> {
+  public async run(): Promise<AccountBase> {
     const config = await Config.init();
     const accountsDomain = await Accounts.initFromFlags(this.flags, config);
-    const accountInfo = await accountsDomain.get(this.args.account!);
+    const accountBaseInfo = await accountsDomain.accountBaseFromAddress(this.args.account!);
 
-    if (!this.jsonEnabled()) {
-      this.warning(
-        `${yellow('Attention:')} this will permanently delete the account ${bold.green(accountInfo.name)} (${green(
-          accountInfo.address
-        )})\n`
-      );
-    }
+    await accountsDomain.assertAccountExists(this.args.account!);
```

Comment:
Please move this assertion to the `remove` method, as those checks should be part of the Accounts domain. Adding it to the `get` method will also be great.

Reply:
Thanks for the review, this check is needed much before we attempt to delete, we need it before we ask for confirmation.
So now created a new function in the Accounts domain that contains this check when getting the AccountBase data and has more specific logic for this scenario